src/Chapter4.hs

```diff
@@ -629,7 +659,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM a b = a >>= (\x -> if x then b >>= (\y -> pure (x && y)) else pure x)
```

Comment:
That's correct implementation! Though, it could be simplified a bit, knowing the fact that `True && x` is always `x`.

```suggestion
andM a b = a >>= (\x -> if x then b else pure False)
```

Reply:
AH!!!!! thanks (: